vip-helpers/vip-caching.php

```diff
@@ -188,7 +188,11 @@ function wpcom_vip_get_page_by_path( $page_path, $output = OBJECT, $post_type =
 	if ( $page_id === false ) {
 		$page = get_page_by_path( $page_path, $output, $post_type );
 		$page_id = $page ? $page->ID : 0;
-		wp_cache_set( $cache_key, $page_id, 'get_page_by_path' ); // We only store the ID to keep our footprint small
+		if ( $page_id ===0 ){
```

Comment:
This is a direct copy from WordPress.com SVN.  I'd be happy to send it through PHPCS and have it auto-fix whitespace issues.